[24.2] Move invocation view running actions and improve styling of annotation section

ahmedhamidawan opened the pull request:
The actions for a currently running workflow were added in the `tabs-end` slot in #18615  Here, we remove controls from there and the `BAlert` and instead:
- The icon on the top left is a loading one when workflow is running
- The cancel button is ~~next to the progress bars~~ now in the title actions (next to the Play button)
- We remove the "Generate PDF" button entirely

Fixes https://github.com/galaxyproject/galaxy/issues/19158

Also lets the workflow description/annotation take up full width, and adds a `<hr>` below it to separate from the run form.

Fixes https://github.com/galaxyproject/galaxy/issues/19157

| Before 🟥  |
| ---- |
| ![invocation_controls_fix_BEFORE](https://github.com/user-attachments/assets/c25b0b21-0ce0-43d2-97ab-45ec7835501d) | 

| Now ✅ |
| ---- |
| ![invocation_controls_fix_AFTER](https://github.com/user-attachments/assets/4d7fb719-aa38-4520-94e9-225762eca596) |

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
Move invocation view controls and loading indicator

Used to be in the `tabs-end` slot. Here, we remove it from there and the BAlert and instead:
- The icon on the top left is a loading one when workflow is running
- The cancel button is next to the progress bars
- We remove the "Generate PDF" button entirely

Fixes https://github.com/galaxyproject/galaxy/issues/19158

martenson commented on client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue:
```diff
@@ -272,6 +274,17 @@ function cancelWorkflowSchedulingLocal() {
+                    data-description="cancel invocation button"
+                    size="sm"
+                    pill
+                    variant="outline-danger"
```
I'd suggest "downgrading" the visual language. The operation is not dangerous, you lose nothing.
```suggestion
                    variant="outline-warning"
```

ahmedhamidawan replied:
That would be this:
| Current `outline-danger` |
| ---- |
| ![firefox_zsCj3vgJBD](https://github.com/user-attachments/assets/7e0e60b6-dfe6-4d87-bebc-7fe67e5a3a90) |



| Changed to `outline-warning` |
| ---- |
| ![firefox_DusLmqgIYP](https://github.com/user-attachments/assets/55018684-a741-4402-b67b-01e4d3b0abad) |

**What do we think?** 🤔
React 👍 for changing to warning, 👎 for keeping danger

mvdbeek replied:
Both of these don't match our style, also that button is way too big. Have you considered making it a normal stop button next to the play button ?

ahmedhamidawan replied:
![image](https://github.com/user-attachments/assets/f34125ad-52e4-4273-9fc1-f56dd8d83fa0)

What about this?
The icon can be kept as the ✖️ instead, and the text can be `Cancel` instead of `Stop`
whatever works...

mvdbeek replied:
Nice, I like it!

itisAliRH commented on client/src/components/Workflow/WorkflowNavigationTitle.vue:
```diff
@@ -102,7 +103,8 @@ const workflowImportTitle = computed(() => {
                     <div class="flex-grow-1" data-description="workflow heading">
                         <div class="px-1">
-                            <FontAwesomeIcon :icon="faSitemap" />
+                            <FontAwesomeIcon v-if="props.invocationRunning" :icon="faSpinner" spin />
```
Since we mainly use the spinner icon as a loading/fetching indicator, I'd recommend using the spinning gear icon instead here

ahmedhamidawan replied:
Do we use the gear icon anywhere? Could be the first here for showing a running operation... I wonder if we have some example somewhere for an icon for something currently running?

Also, I did think about making the workflow (faSitemap) icon spin as well 😅🤷‍♂️?

ahmedhamidawan replied:
| Current `faSpinner` |
| ---- |
| ![firefox_hq8OtFiFTZ](https://github.com/user-attachments/assets/db722f83-5b15-48f0-9fb9-a00276a029f7) |

| `faCog` |
| ---- |
| ![firefox_en3AmtlMKY](https://github.com/user-attachments/assets/451878ed-f82e-48fb-8cf5-d44b17e6f737) |

| `faSitemap` |
| ---- |
| ![firefox_YBz3W6uJgZ](https://github.com/user-attachments/assets/62940750-12e7-48ef-b915-a35a6fc73dff) |

dannon replied:
We use the same spinner for running jobs, and it has a reasonable historical meaning here—I don't mind it in this context.

I am against the cog here. While it looks cool, we already use it for a completely different purpose (settings, options, etc.), as is the common usage these days and historically in Galaxy.

faSitemap is hilarious—maybe for April Fools.

All that said, I'm not sure anything needs to be spinning here—the rest of this interface makes it pretty clear when the workflow is still running with the bars, I think.

ahmedhamidawan replied:
Done. Removed the spinner entirely..

ahmedhamidawan committed:
reintroduce `tabs-end` slot just for the invocation grid for cancel run

In the grid version, we don't have the `WorkflowNavigationTitle` and `WorkflowAnnotation` components, and one of those has the workflow cancel button, which could be useful in the grid as well.

ahmedhamidawan committed:
only show `SwitchToHistoryLink` in invocation view

This reverts the change in commit 632955ab5d118f0c8862b225e3d4648b08ff04a3
We realized (in https://github.com/galaxyproject/galaxy/issues/19157) we didn't really need this in the run form version of `WorkflowAnnotation` because there this acts as an indicator of whatever the "input" history is; which is just your current history and users can see that anyways.

ahmedhamidawan committed:
move cancel invocation button to `workflow-title-actions`

ahmedhamidawan committed:
fix `TextSummary` so if `one-line-summary="true"`, expansion works

ahmedhamidawan committed:
use `one-line-summary` version of `TextSummary` and adds an `<hr>`

Fixes https://github.com/galaxyproject/galaxy/issues/19157 where we notice that the description could be full width instead of truncated half way, as well as the need for a visual distinction between the run form and the header annotation.

ahmedhamidawan committed:
remove success message from `WorkflowRunSuccess` in case of 1 run

and add it to `WorkflowNavigationTitle` where it fades out to reveal the title and actions

ahmedhamidawan committed:
fix selector for report tab in jest for invocation view

ahmedhamidawan committed:
jest checks for `SwitchToHistoryLink`'s conditional existence

which is based on whether it is the run form or invocation view

Tests changes from commit 6dd5f7c4c76b97c27f650d920839ab9deed02e5d

ahmedhamidawan committed:
remove workflow running spinner from invocation view

since the progress bars already indicate that

mvdbeek reviewed (approved):
Nice, thank you!
